### PR TITLE
REST API to recalculate bits and clone katalogus settings

### LINKED ISSUE #3525: Create REST API to rerun bits and duplicate katalogus settings
- Create REST API that will rerun the bits for the specified organization.
- Duplicate boefje settings from one organisation to another organisation

### PR DESCRIPTION
**Pull request by dekkers**
### Changes

This adds the REST API for recalculating bits and cloning settings. The endpoints are under the organization endpoint. For example if organization has rocky  primary key 1, using httpie you can recalculate bits using:

```bash
$ http POST http://127.0.0.1:8000/api/v1/organization/1/recalculate_bits/ "Authorization:Token <token>"
```

To clone the katalogus settings to organization with rocky primary key 2:

```bash
$ http POST http://127.0.0.1:8000/api/v1/organization/1/clone_katalogus_settings/ to_organization=2 "Authorization:Token <token>"
```

This includes PR #3571 because else the tests with the right permissions don't work, so PR #3571 needs to be merged first.

### Issue link

Closes #3525 

### QA notes

See Changes how to test the API.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — DISMISSED** [review]
## Checklist for QA:

- [ ] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [ ] I have managed to pass the onboarding flow
  - [ ] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:

- API endpoints exist and work as expected
- Was able to recalculate bits
- Was able to clone KATalogus settings to a new organization

File: rocky/tools/viewsets.py [hunk 1]
```diff
@@ -40,3 +54,24 @@ def set_indemnification(self, request, pk=None):
         indemnification = Indemnification.objects.create(organization=organization, user=self.request.user)
 
         return Response({"indemnification": True, "user": indemnification.user.pk}, status=status.HTTP_201_CREATED)
+
+    @action(detail=True, methods=["post"], permission_classes=[CanRecalculateBits])
+    def recalculate_bits(self, request, pk=None):
+        organization = self.get_object()
+        connector = OctopoesAPIConnector(settings.OCTOPOES_API, organization.code)
+        number_of_bits = connector.recalculate_bits()
+
```
Comment (underdarknl) [inline rocky/tools/viewsets.py:63]: We need to emit an audit-event here.

File: rocky/tools/viewsets.py [hunk 2]
```diff
@@ -40,3 +54,24 @@ def set_indemnification(self, request, pk=None):
...
+    @action(detail=True, methods=["post"], permission_classes=[CanSetKatalogusSettings])
+    def clone_katalogus_settings(self, request, pk=None):
+        from_organization = self.get_object()
+
+        serializer = ToOrganizationSerializer(data=request.data)
+        if serializer.is_valid():
+            to_organization = serializer.validated_data["to_organization"]
+            get_katalogus(from_organization.code).clone_all_configuration_to_organization(to_organization.code)
+
```
Comment (underdarknl) [inline rocky/tools/viewsets.py:74]: We need to emit an audit-event here.

**Review by underdarknl — APPROVED** [review]

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
  - [ ] I have managed to pass the onboarding flow
  - [ ] Objects and Findings are created properly
  - [ ] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [ ] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Couldn't test due to the error message below. 

### What doesn't work:
- When going to the API interface the following error is triggered when using the form to recalculate the bits and to clone. 

image
 

### Bug or feature?:
n/a

File: rocky/tools/viewsets.py [hunk 3]
```diff
@@ -50,3 +57,31 @@ def set_indemnification(self, request, pk=None):
...
+    def clone_katalogus_settings(self, request, pk=None):
+        from_organization = self.get_object()
+
+        serializer = ToOrganizationSerializer(data=request.data)
+        if serializer.is_valid():
+            to_organization = serializer.validated_data["to_organization"]
+            logger.info(
+                "Cloning organization settings",
+                event_code=910000,
+                organization_code=self.organization.code,
```
[the remark this answers is not in the document]
Comment (underdarknl) [reply rocky/tools/viewsets.py:80]: This should fi the 500 that @stephanie0x00 noticed, and is also triggered in the UnitTests

**Review by underdarknl — APPROVED** [review]

**Pull request merged**